### PR TITLE
Add all_match, any_match and none_match functions for arrays

### PR DESCRIPTION
**Pull request by wenleix**
Backported from https://github.com/prestosql/presto/pull/1045

```
== RELEASE NOTES ==

General Changes
* Add all_match(), any_match(), and none_match() functions.
```

### CODE REVIEW

**Comment by wenleix** [comment]
cc @kaikalur

File: presto-docs/src/main/sphinx/functions/array.rst [hunk 1]
```diff
@@ -21,6 +21,20 @@ The ``||`` operator is used to concatenate an array with an array or an element
 Array Functions
 ---------------
 
+.. function:: all_match(array(T), function(T,boolean)) -> boolean
```
Comment (rongrong) [inline presto-docs/src/main/sphinx/functions/array.rst]: Space after `T,` before `boolean` => `T, boolean`

File: presto-docs/src/main/sphinx/functions/array.rst [hunk 2]
```diff
@@ -21,6 +21,20 @@ The ``||`` operator is used to concatenate an array with an array or an element
...
+
+    Returns whether all elements of an array match the given predicate. Returns ``true`` if all the elements
+    match the predicate (a special case is when the array is empty); ``false`` if one or more elements don't
+    match; ``NULL`` if the predicate function returns ``NULL`` for one or more elements and ``true`` for all
+    other elements.
+
+.. function:: any_match(array(T), function(T,boolean)) -> boolean
```
Comment (rongrong) [inline presto-docs/src/main/sphinx/functions/array.rst]: same as above

File: presto-docs/src/main/sphinx/functions/array.rst [hunk 3]
```diff
@@ -21,6 +21,20 @@ The ``||`` operator is used to concatenate an array with an array or an element
 Array Functions
 ---------------
 
+.. function:: all_match(array(T), function(T,boolean)) -> boolean
+
+    Returns whether all elements of an array match the given predicate. Returns ``true`` if all the elements
+    match the predicate (a special case is when the array is empty); ``false`` if one or more elements don't
+    match; ``NULL`` if the predicate function returns ``NULL`` for one or more elements and ``true`` for all
+    other elements.
```
Comment (rongrong) [inline presto-docs/src/main/sphinx/functions/array.rst]: What about empty array? Every predicate would return true? Is that worth documenting?

File: presto-docs/src/main/sphinx/functions/array.rst [hunk 4]
```diff
@@ -139,6 +153,12 @@ Array Functions
         SELECT ngrams(ARRAY['foo', 'bar', 'baz', 'foo'], 5); -- [['foo', 'bar', 'baz', 'foo']]
         SELECT ngrams(ARRAY[1, 2, 3, 4], 2); -- [[1, 2], [2, 3], [3, 4]]
 
+.. function:: none_match(array(T), function(T,boolean)) -> boolean
```
Comment (rongrong) [inline presto-docs/src/main/sphinx/functions/array.rst]: It doesn't seem that the functions are actually listed in alphabetical order so it probably makes more sense to keep these three functions together.

File: presto-docs/src/main/sphinx/functions/array.rst (hunk 3 above)
Comment (wenleix) [reply presto-docs/src/main/sphinx/functions/array.rst]: @rongrong : It's already discussed:
```
(a special case is when the array is empty)
```

File: presto-docs/src/main/sphinx/functions/array.rst (hunk 4 above)
Comment (wenleix) [reply presto-docs/src/main/sphinx/functions/array.rst]: @rongrong : Looks to me the functions are listed in alphabetical order? Maybe some functions are not following the ordering strictly (although I didn't see it with skim 😄 ) and we should fix them?

Comment (rongrong) [reply presto-docs/src/main/sphinx/functions/array.rst]: Maybe it's the array_intersect/union/except that's wrong then. Can you add a follow up PR to fix that? Thanks!

Comment (rongrong) [reply presto-docs/src/main/sphinx/functions/array.rst]: also add space here.

Comment (wenleix) [reply presto-docs/src/main/sphinx/functions/array.rst]: @rongrong 
> Maybe it's the array_intersect/union/except that's wrong then. Can you add a follow up PR to fix that? Thanks!

Yup: https://github.com/prestodb/presto/pull/13736 😃

**Pull request merged**